pkg/config/constants.go

```diff
@@ -177,84 +177,84 @@ const DefaultPrometheusConfiguration = `
 lowercaseOutputName: true
 attrNameSnakeCase: true
 rules:
-  - pattern: metrics<name=(\S+)-(\S+)\.driver\.(BlockManager|DAGScheduler|jvm)\.(\S+)><>Value
+  - pattern: metrics<name=(\S+).(\S+)\.driver\.(BlockManager|DAGScheduler|jvm)\.(\S+)><>Value
     name: spark_driver_$3_$4
     type: GAUGE
     labels:
       app_namespace: "$1"
       app_name: "$2"
-  - pattern: metrics<name=(\S+)-(\S+)\.driver\.(\S+)\.StreamingMetrics\.streaming\.(\S+)><>Value
+  - pattern: metrics<name=(\S+).(\S+)\.driver\.(\S+)\.StreamingMetrics\.streaming\.(\S+)><>Value
     name: spark_streaming_driver_$4
     type: GAUGE
     labels:
       app_namespace: "$1"
```

Comment:
The app_namespace prefix needs to be removed. Spark core metrics does not have this.
The pattern is like this: https://github.com/prometheus/jmx_exporter/blob/master/example_configs/spark.yml#L14

Reply:
@liyinan926 Are you ok with me removing the namespace prefix from metric names altogether?

Reply:
I prefer keeping them. Even though Spark core metrics in the linked example don't have namespace, we can always support them by adding the namespace as part of the prefix. It's useful to filter metrics by namespaces.

Reply:
@liyinan926 Ok. Could you review this PR and get it merged? Thanks!

Reply:
@liyinan926 it does not make sense to add it. Spark core emits these metrics. How would you modify them? This right now creates problems with parsing.

Reply:
You can specify prefix for metrics using `spark.metrics.namespace`.

Reply:
Yes but the default shouldnt be that IMHO.